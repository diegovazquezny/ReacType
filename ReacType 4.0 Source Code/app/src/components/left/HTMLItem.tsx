import React, { useState } from 'react';
import { useDrag } from 'react-dnd';
import { ItemTypes } from '../../constants/ItemTypes';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import createModal from '../right/createModal';

const buttonClasses =
'MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-12 MuiButton-textPrimary';

const useStyles = makeStyles({
  HTMLPanelItem: {
    color: 'white',
    // this is experimental for version: BLADERUNNER THEME
    backgroundColor: 'transparent',
    // minWidth: '340px',
    minHeight: '60px',
    marginBottom: '10px',
    marginRight: '5px',
    marginLeft: '5px',
    border: '2px dotted rgba(255,255,255, 0.45)',
    borderRadius: '8px',
    cursor: 'grab',
    '& > h3': {
      display: 'inline-block',
      paddingTop: '18px'
    }
  }
});

const HTMLItem: React.FC<{
  name: string;
  id: number;
  Icon: any;
  handleDelete: (id: number) => void;
}> = ({ name, id, Icon, handleDelete }) => {
  const classes = useStyles();
  const [modal, setModal] = useState(null);
  const [{ isDragging }, drag] = useDrag({
    item: {
      type: ItemTypes.INSTANCE,
      newInstance: true,
      instanceType: 'HTML Element',
      name,
      instanceTypeId: id
    },
    collect: (monitor: any) => ({
      isDragging: !!monitor.isDragging()
    })
  });

  const closeModal = () => setModal(null);

  // creates modal that asks if user wants to clear workspace
  // if user clears their workspace, then their components are removed from state and the modal is closed
  const deleteAllInstances = (deleteID: number) => {
    // set modal options
    const children = (
      <List className="export-preference">
        <ListItem
          key={`clear${deleteID}`}
          button
          onClick={() => handleDelete(deleteID)}
          style={{
            border: '1px solid #3f51b5',
            marginBottom: '2%',
            marginTop: '5%'
          }}
        >
          <ListItemText
            primary={'Yes, delete all instances'}
            style={{ textAlign: 'center' }}
            onClick={closeModal}
          />
        </ListItem>
        <ListItem
          key={`close${deleteID}`}
          button
          onClick={closeModal}
          style={{
            border: '1px solid #3f51b5',
            marginBottom: '2%',
            marginTop: '5%'
          }}
        >
          <ListItemText
            primary={'No, do not delete element'}
            style={{ textAlign: 'center' }}
            onClick={closeModal}
          />
        </ListItem>
      </List>
    );

    // create modal
    setModal(
      createModal({
        closeModal,
        children,
        message: 'Deleting this element will delete all instances of this element within the application.\nDo you still wish to proceed?',
        primBtnLabel: null,
        primBtnAction: null,
        secBtnAction: null,
        secBtnLabel: null,
        open: true
      })
    );
  };

  return (
    <Grid item xs={5} key={`html-g${name}`}>
      <div ref={drag} className={classes.HTMLPanelItem}>
        <h3>{name}</h3>
        <span
          style={{
            verticalAlign: 'middle',
            display: 'inline-block',
            marginLeft: '5px'
          }}
        >
          {Icon && <Icon />}
        </span>
        {id > 11 &&
        <button className={buttonClasses} onClick={() => deleteAllInstances(id)} > X </button> }
      </div>
      {modal}
    </Grid>
  );
}

export default HTMLItem;
