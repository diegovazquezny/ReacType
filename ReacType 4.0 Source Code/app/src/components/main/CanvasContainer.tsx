import React from 'react';
import Canvas from './Canvas';

// The CanvasContainer sets the boundaries on the width/height of the canvas
function CanvasContainer() {
  const canvasContainerStyle = {
    width: '100%',
    backgroundColor: 'lightgrey',
    border: '2px Solid grey',
  };

  return (
    <div style={canvasContainerStyle}>
      <Canvas />
    </div>
  );
}

export default CanvasContainer;
