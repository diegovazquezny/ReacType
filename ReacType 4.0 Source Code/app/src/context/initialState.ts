import { State } from '../interfaces/Interfaces';
import HTMLTypes from './HTMLTypes';

// This is the state that will be set for a new project or when the user resets the state of their project
const initialState: State = {
  name: '',
  isLoggedIn: false,
  components: [
    {
      id: 1,
      name: 'index',
      style: {},
      code: '<div>Drag in a component or HTML element into the canvas!</div>',
      children: [],
      isPage: true
    }
  ],
  projectType: 'Next.js',
  rootComponents: [1],
  canvasFocus: { componentId: 1, childId: null },
  nextComponentId: 2,
  nextChildId: 1,
  HTMLTypes
};

export default initialState;
