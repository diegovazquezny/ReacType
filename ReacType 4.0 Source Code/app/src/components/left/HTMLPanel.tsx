import React, { useState, useContext } from 'react';
import Grid from '@material-ui/core/Grid';
import StateContext from '../../context/context';
import HTMLItem from './HTMLItem';
import { makeStyles } from '@material-ui/core/styles';

const HTMLPanel = (): JSX.Element => {
  const classes = useStyles();
  const [tag, setTag] = useState('');
  const [name, setName] = useState('');
  const [errorMsg, setErrorMsg] = useState('');
  const [errorStatus, setErrorStatus] = useState(false);
  const [state, dispatch] = useContext(StateContext);

  let startingID = 0;
  state.HTMLTypes.forEach(element => {
    if (element.id >= startingID) startingID = element.id;
  });
  startingID += 1;

  const [currentID, setCurrentID] = useState(startingID);

  const buttonClasses =
    'MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-12 MuiButton-textPrimary';

  const handleTagChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    resetError();
    setTag(e.target.value);
  };

  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    resetError();
    setName(e.target.value);
  };

  const checkNameDupe = (inputName: String): boolean => {
    let checkList = state.HTMLTypes.slice();

    // checks to see if inputted comp name already exists
    let dupe = false;
    checkList.forEach(HTMLTag => {
      if (
        HTMLTag.name.toLowerCase() === inputName.toLowerCase() ||
        HTMLTag.tag.toLowerCase() === inputName.toLowerCase()
      ) {
        dupe = true;
      }
    });
    return dupe;
  };

  const triggerError = (type: String) => {
    setErrorStatus(true);
    if (type === 'empty') {
      setErrorMsg('Tag/ Tag name cannot be blank.');
    } else if (type === 'dupe') {
      setErrorMsg('Tag/ Tag name already exists.');
    } else if (type === 'letters') {
      setErrorMsg('Tag/ Tag name must start with a letter.');
    } else if (type === 'symbolsDetected') {
      setErrorMsg('Tag/ Tag name must not contain symbols.');
    }
  };

  const resetError = () => {
    setErrorStatus(false);
  };

  const createOption = (inputTag: String, inputName: String) => {
    // format name so first letter is capitalized and there are no whitespaces
    let inputNameClean = inputName.replace(/\s+/g, '');
    const formattedName =
      inputNameClean.charAt(0).toUpperCase() + inputNameClean.slice(1);
    // add new component to state
    const newElement = {
      id: currentID,
      tag: inputTag,
      name: formattedName,
      style: {},
      placeHolderShort: name,
      placeHolderLong: '',
      icon: null
    };
    dispatch({
      type: 'ADD ELEMENT',
      payload: newElement
    });
    setCurrentID(currentID + 1);
    setTag('');
    setName('');
  };

  const alphanumeric = (input: string): boolean => {
    let letterNumber = /^[0-9a-zA-Z]+$/;
    if (input.match(letterNumber)) return true;
    return false;
  };

  const handleSubmit = e => {
    e.preventDefault();
    let letters = /[a-zA-Z]/;
    if (!tag.charAt(0).match(letters) || !name.charAt(0).match(letters)) {
      triggerError('letters');
      return;
    } else if (!alphanumeric(tag) || !alphanumeric(name)) {
      triggerError('symbolsDetected');
      return;
    } else if (tag.trim() === '' || name.trim() === '') {
      triggerError('empty');
      return;
    } else if (checkNameDupe(tag) || checkNameDupe(name)) {
      triggerError('dupe');
      return;
    }
    createOption(tag, name);
    resetError();
  };

  const handleDelete = (id: number): void => {
    dispatch({
      type: 'DELETE ELEMENT',
      payload: id
    });
  };

  return (
    <div>
      <h4> HTML Elements</h4>
      <div className={classes.addComponentWrapper}>
        <div className={classes.inputWrapper}>
          <form onSubmit={handleSubmit}>
            <h4>Create New Element: </h4>
            <label className={classes.inputLabel}>
              Tag:
              <input
                color={'primary'}
                type="text"
                name="Tag"
                value={tag}
                onChange={handleTagChange}
                className={classes.input}
                style={{ marginBottom: '10px' }}
              />
              {errorStatus && <span>{errorMsg}</span>}
            </label>
            <br></br>
            <label className={classes.inputLabel}>
              Tag Name:
              <input
                color={'primary'}
                type="text"
                name="Tag Name"
                value={name}
                onChange={handleNameChange}
                className={classes.input}
              />
              {errorStatus && <span>{errorMsg}</span>}
            </label>
            <input
              className={buttonClasses}
              color="primary"
              type="submit"
              value="Add Element"
              style={{ marginTop: '15px' }}
            />
          </form>
        </div>
      </div>
      <Grid
        container
        spacing={1}
        direction="row"
        justify="center"
        alignItems="center"
      >
        {state.HTMLTypes.map(option => (
          <HTMLItem
            name={option.name}
            key={`html-${option.name}`}
            id={option.id}
            Icon={option.icon}
            handleDelete={handleDelete}
          />
        ))}
      </Grid>
    </div>
  );
};

const useStyles = makeStyles({
  inputWrapper: {
    // height: '115px',
    textAlign: 'center',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    // paddingLeft: '35px',
    marginBottom: '15px'
  },
  addComponentWrapper: {
    border: '1px solid rgba(70,131,83)',
    padding: '20px',
    margin: '20px'
  },
  input: {
    color: '#fff',
    borderRadius: '5px',
    paddingLeft: '15px',
    paddingRight: '10px',
    whiteSpace: 'nowrap',
    overflowX: 'hidden',
    textOverflow: 'ellipsis',
    border: '1px solid rgba(51,235,145,0.75)',
    backgroundColor: 'rgba(255,255,255,0.15)',
    marginLeft: '10px'
  },
  inputLabel: {
    fontSize: '16px',
    zIndex: 20,
    color: '#fff',
    marginTop: '-10px'
  }
});

export default HTMLPanel;
