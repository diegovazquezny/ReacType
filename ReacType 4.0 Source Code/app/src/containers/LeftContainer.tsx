import React, { useContext } from 'react';
import Grid from '@material-ui/core/Grid';
import ComponentPanel from '../components/left/ComponentPanel';
import HTMLPanel from '../components/left/HTMLPanel';
import { styleContext } from './AppContainer';

// Left-hand portion of the app, where component options are displayed
const LeftContainer = (): JSX.Element => {
  const { style } = useContext(styleContext);

  return (
    <div className="column left" style={style}>
      <Grid container direction="row" alignItems="center">
        <ComponentPanel />
        <HTMLPanel />
      </Grid>
    </div>
  );
};

export default LeftContainer;
