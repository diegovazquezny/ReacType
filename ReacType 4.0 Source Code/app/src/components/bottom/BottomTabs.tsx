import React, { useState, useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import StateContext from '../../context/context';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import CodePreview from './CodePreview';
import Box from '@material-ui/core/Box';
import Tree from '../../tree/TreeChart';
import { emitKeypressEvents } from 'readline';
import NativeSelect from '@material-ui/core/NativeSelect';
import FormControl from '@material-ui/core/FormControl';
import { styleContext } from '../../containers/AppContainer';

const BottomTabs = () => {
  // state that controls which tab the user is on
  const [state, dispatch] = useContext(StateContext);
  const [tab, setTab] = useState(0);
  const classes = useStyles();
  treeWrapper: HTMLDivElement;
  const [theme, setTheme] = useState('monokai');
  const { style } = useContext(styleContext);

  // method changes the
  const handleChange = (event: React.ChangeEvent, value: number) => {
    setTab(value);
  };

  const { components, HTMLTypes } = state;

  const changeTheme = e => {
    setTheme(e.target.value);
  };

  return (
    <div className={classes.root} style={style}>
      <Box display="flex" justifyContent="space-between">
        <Tabs
          value={tab}
          onChange={handleChange}
          classes={{
            root: classes.tabsRoot,
            indicator: classes.tabsIndicator
          }}
        >
          <Tab
            disableRipple
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
            label="Code Preview"
          />
          <Tab
            disableRipple
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
            label="Component Tree"
          />
        </Tabs>
        <FormControl>
          <div className="flex-container">
            <div className="flex1">Change Theme:</div>
            <NativeSelect
              className="flex2"
              style={{ color: 'white' }}
              value={theme}
              onChange={changeTheme}
            >
              <option style={{ backgroundColor: '#252526' }} value={'monokai'}>
                Monokai
              </option>
              <option style={{ backgroundColor: '#252526' }} value={'github'}>
                Github
              </option>
              <option
                style={{ backgroundColor: '#252526' }}
                value={'solarized_dark'}
              >
                Solarized Dark
              </option>
              <option style={{ backgroundColor: '#252526' }} value={'terminal'}>
                Terminal
              </option>
              <option
                style={{ backgroundColor: '#252526' }}
                value={'solarized_light'}
              >
                Solarized Light
              </option>
            </NativeSelect>
          </div>
        </FormControl>
      </Box>
      {tab === 0 && <CodePreview theme={theme} setTheme={setTheme} />}
      {tab === 1 && <Tree data={components} />}
    </div>
  );
};

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: '#333333',
    height: '100%',
    color: '#fff',
    boxShadow: '0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)'
  },
  bottomHeader: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    Width: '200px'
  },
  tabsRoot: {
    borderBottom: '0.5px solid #424242',
    minHeight: '50%'
  },
  tabsIndicator: {
    backgroundColor: '#1de9b6'
  },
  tabRoot: {
    textTransform: 'initial',
    minWidth: 40,
    fontWeight: theme.typography.fontWeightRegular,
    marginRight: theme.spacing(4), // JZ: updated syntax as per deprecation warning

    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      '"Helvetica Neue"',
      'Arial',
      'sans-serif',
      '"Apple Color Emoji"',
      '"Segoe UI Emoji"',
      '"Segoe UI Symbol"'
    ].join(','),
    '&:hover': {
      color: '#1de9b6',
      opacity: 1
    },
    '&$tabSelected': {
      color: '#33eb91',
      fontWeight: theme.typography.fontWeightMedium
    },
    '&:focus': {
      color: '#4aedc4'
    }
  },
  tabSelected: {},
  typography: {
    padding: theme.spacing(3)
  },
  padding: {
    padding: `0 ${theme.spacing(2)}px`
  },
  switch: {
    marginRight: '10px',
    marginTop: '2px'
  }
}));

export default BottomTabs;
