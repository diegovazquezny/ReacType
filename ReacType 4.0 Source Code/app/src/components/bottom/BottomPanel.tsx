import React, { useContext } from 'react';
// import StateContext from '../../context/context';
import BottomTabs from './BottomTabs';
import { Resizable } from 're-resizable';


// const IPC = require('electron').ipcRenderer;

const BottomPanel = () => {
  return (
    <Resizable
      enable={{
        top: true
      }}
      minHeight={'25%'}
    >
      <div className="bottom-panel">
        <BottomTabs  />
      </div>
    </Resizable>
  );
};

export default BottomPanel;
