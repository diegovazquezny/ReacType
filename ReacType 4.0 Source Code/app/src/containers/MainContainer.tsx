import React, { useContext } from 'react';
import BottomPanel from '../components/bottom/BottomPanel';
import CanvasContainer from '../components/main/CanvasContainer';
import { styleContext } from './AppContainer';

const MainContainer = () => {
  const { style } = useContext(styleContext);
  return (
    <div className="main-container" style={style} >
      <div className="main">
        <CanvasContainer />
      </div>
      <BottomPanel />
    </div>
  );
};

export default MainContainer;
