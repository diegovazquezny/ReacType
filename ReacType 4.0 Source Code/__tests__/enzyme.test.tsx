import { shallow, mount } from 'enzyme';
import React, { useState, useContext } from 'react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import StateContext from '../app/src/context/context';
import initialState from '../app/src/context/initialState';
import MainContainer from '../app/src/containers/MainContainer';
import BottomPanel from '../app/src/components/bottom/BottomPanel';
import BottomTabs from '../app/src/components/bottom/BottomTabs';
import CodePreview from '../app/src/components/bottom/CodePreview';
import CanvasContainer from '../app/src/components/main/CanvasContainer';
import Canvas from '../app/src/components/main/Canvas';
import HTMLPanel from '../app/src/components/left/HTMLPanel';
import HTMLItem from '../app/src/components/left/HTMLItem';
import LeftContainer from '../app/src/containers/LeftContainer';

// npm test -- -u

describe('Test the CanvasContainer component', () => {
  const target = shallow(<CanvasContainer />);
  it('Matches snapshot', () => {
    expect(target).toMatchSnapshot();
  });
  it('Contains Canvas component', () => {
    expect(target.contains(<Canvas />)).toBe(true);
  });
});

describe('Test the MainContainer component', () => {
  const target = shallow(<MainContainer />);
  it('Contains CanvasContainer component', () => {
    expect(target.contains(<CanvasContainer />)).toBe(true);
  });
  it('Contains BottomPanel component', () => {
    expect(target.contains(<BottomPanel />)).toBe(true);
  });
});

describe('Test the BottomTabs component', () => {
  const target = shallow(<BottomTabs />);
  it('Matches snapshot', () => {
    expect(target).toMatchSnapshot();
  });
});

describe('Test All 10 default HTML elements have rendered', () => {
  const target = shallow(
    <DndProvider backend={HTML5Backend}>
      <StateContext.Provider value={initialState}>
        <HTMLPanel />
      </StateContext.Provider>
    </DndProvider>
  );

  it('Matches snapshot', () => {
    expect(target).toMatchSnapshot();
  });
});
