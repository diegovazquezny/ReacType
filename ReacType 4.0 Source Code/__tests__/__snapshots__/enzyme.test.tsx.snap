// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test All 10 default HTML elements have rendered Matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "dragDropManager": DragDropManagerImpl {
        "backend": HTML5BackendImpl {
          "actions": Object {
            "beginDrag": [Function],
            "drop": [Function],
            "endDrag": [Function],
            "hover": [Function],
            "publishDragSource": [Function],
          },
          "altKeyPressed": false,
          "asyncEndDragFrameId": null,
          "currentDragSourceNode": null,
          "currentNativeHandle": null,
          "currentNativeSource": null,
          "dragEnterTargetIds": Array [],
          "dragOverTargetIds": null,
          "dragStartSourceIds": null,
          "dropTargetIds": Array [],
          "endDragIfSourceWasRemovedFromDOM": [Function],
          "endDragNativeItem": [Function],
          "enterLeaveCounter": EnterLeaveCounter {
            "entered": Array [],
            "isNodeInDocument": [Function],
          },
          "getSourceClientOffset": [Function],
          "handleSelectStart": [Function],
          "handleTopDragEndCapture": [Function],
          "handleTopDragEnter": [Function],
          "handleTopDragEnterCapture": [Function],
          "handleTopDragLeaveCapture": [Function],
          "handleTopDragOver": [Function],
          "handleTopDragOverCapture": [Function],
          "handleTopDragStart": [Function],
          "handleTopDragStartCapture": [Function],
          "handleTopDrop": [Function],
          "handleTopDropCapture": [Function],
          "isNodeInDocument": [Function],
          "monitor": DragDropMonitorImpl {
            "registry": HandlerRegistryImpl {
              "dragSources": Map {},
              "dropTargets": Map {},
              "pinnedSource": null,
              "pinnedSourceId": null,
              "store": Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              },
              "types": Map {},
            },
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "mouseMoveTimeoutTimer": null,
          "options": OptionsReader {
            "globalContext": [Window],
          },
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "sourceNodeOptions": Map {},
          "sourceNodes": Map {},
          "sourcePreviewNodeOptions": Map {},
          "sourcePreviewNodes": Map {},
        },
        "handleRefCountChange": [Function],
        "isSetUp": false,
        "monitor": DragDropMonitorImpl {
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
    }
  }
>
  <ContextProvider
    value={
      Object {
        "HTMLTypes": Array [
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "TextFormatIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 11,
            "name": "Div",
            "placeHolderLong": "",
            "placeHolderShort": "",
            "style": Object {},
            "tag": "div",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "ImageIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 1,
            "name": "Image",
            "placeHolderLong": "",
            "placeHolderShort": "Image",
            "style": Object {},
            "tag": "img",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "DescriptionIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 2,
            "name": "Form",
            "placeHolderLong": "",
            "placeHolderShort": "Form",
            "style": Object {},
            "tag": "form",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "ListIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 3,
            "name": "List",
            "placeHolderLong": "",
            "placeHolderShort": <ul>
              <li>
                item 1
              </li>
              <li>
                item 2
              </li>
              <li>
                item 3
              </li>
            </ul>,
            "style": Object {
              "color": "purple",
            },
            "tag": "li",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "EditAttributesIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 4,
            "name": "Button",
            "placeHolderLong": "",
            "placeHolderShort": <button>
              Button
            </button>,
            "style": Object {
              "border": "none",
              "textAlign": "center",
            },
            "tag": "button",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "LinkIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 6,
            "name": "Link",
            "placeHolderLong": "",
            "placeHolderShort": <a
              href=""
            >
              Link
            </a>,
            "style": Object {
              "border": "none",
            },
            "tag": "a",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "LocalParkingIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 8,
            "name": "Paragraph",
            "placeHolderLong": "",
            "placeHolderShort": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatu",
            "style": Object {},
            "tag": "p",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "TextFormatIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 9,
            "name": "Header 1",
            "placeHolderLong": "",
            "placeHolderShort": "Header 1",
            "style": Object {
              "fontSize": "2em",
            },
            "tag": "h1",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "TextFormatIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 10,
            "name": "Header 2",
            "placeHolderLong": "",
            "placeHolderShort": "Header 2",
            "style": Object {
              "fontSize": "1.5em",
            },
            "tag": "h2",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "TextFormatIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": 5,
            "name": "Span",
            "placeHolderLong": "",
            "placeHolderShort": "Span",
            "style": Object {
              "fontSize": "1.5em",
            },
            "tag": "span",
          },
        ],
        "canvasFocus": Object {
          "childId": null,
          "componentId": 1,
        },
        "components": Array [
          Object {
            "children": Array [],
            "code": "<div>Drag in a component or HTML element into the canvas!</div>",
            "id": 1,
            "isPage": true,
            "name": "index",
            "style": Object {},
          },
        ],
        "isLoggedIn": false,
        "name": "",
        "nextChildId": 1,
        "nextComponentId": 2,
        "projectType": "Next.js",
        "rootComponents": Array [
          1,
        ],
      }
    }
  >
    <HTMLPanel />
  </ContextProvider>
</ContextProvider>
`;

exports[`Test the BottomTabs component Matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
  style={null}
>
  <Styled(MuiBox)
    display="flex"
    justifyContent="space-between"
  >
    <WithStyles(ForwardRef(Tabs))
      classes={
        Object {
          "indicator": "makeStyles-tabsIndicator-4",
          "root": "makeStyles-tabsRoot-3",
        }
      }
      onChange={[Function]}
      value={0}
    >
      <WithStyles(ForwardRef(Tab))
        classes={
          Object {
            "root": "makeStyles-tabRoot-5",
            "selected": "makeStyles-tabSelected-6",
          }
        }
        disableRipple={true}
        label="Code Preview"
      />
      <WithStyles(ForwardRef(Tab))
        classes={
          Object {
            "root": "makeStyles-tabRoot-5",
            "selected": "makeStyles-tabSelected-6",
          }
        }
        disableRipple={true}
        label="Component Tree"
      />
    </WithStyles(ForwardRef(Tabs))>
    <WithStyles(ForwardRef(FormControl))>
      <div
        className="flex-container"
      >
        <div
          className="flex1"
        >
          Change Theme:
        </div>
        <WithStyles(ForwardRef(NativeSelect))
          className="flex2"
          onChange={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
          value="monokai"
        >
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="monokai"
          >
            Monokai
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="github"
          >
            Github
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="solarized_dark"
          >
            Solarized Dark
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="terminal"
          >
            Terminal
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="solarized_light"
          >
            Solarized Light
          </option>
        </WithStyles(ForwardRef(NativeSelect))>
      </div>
    </WithStyles(ForwardRef(FormControl))>
  </Styled(MuiBox)>
  <CodePreview
    setTheme={[Function]}
    theme="monokai"
  />
</div>
`;

exports[`Test the CanvasContainer component Matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "lightgrey",
      "border": "2px Solid grey",
      "width": "100%",
    }
  }
>
  <Canvas />
</div>
`;
